### PR TITLE
include mount information in configreport

### PR DESCRIPTION
**Pull request by phisch**
This PR includes mount information in configreports.
The mount information is added in both the configreport web download and console command reports.

### CODE REVIEW

**Comment by phisch** [comment]
@cdamken please test if this is what you need!

**Comment by cdamken** [comment]
Works for me, but you have remove username and password:

>                 "user": "password",
>                 "password": "password"


Here is the output from before and after:
```
root@cloudine:/var/www/html/owncloud# sudo -u www-data php occ configreport:generate >config1.txt

root@cloudine:/var/www/html/owncloud# cd apps/configreport/

root@cloudine:/var/www/html/owncloud/apps/configreport# patch -p1 < ../../p.patch
patching file lib/AppInfo/Application.php
patching file lib/Command/ConfigReport.php
patching file lib/ReportDataCollector.php

root@cloudine:/var/www/html/owncloud/apps/configreport# cd ../..

root@cloudine:/var/www/html/owncloud# sudo -u www-data php occ configreport:generate >config2.txt

root@cloudine:/var/www/html/owncloud# diff config1.txt config2.txt
4c4
<         "date": "Thu, 13 Apr 2017 12:22:46 +0000",
---
>         "date": "Thu, 13 Apr 2017 12:27:07 +0000",
1847a1848,1872
>         }
>     ],
>     "mounts": [
>         {
>             "id": 1,
>             "mount_point": "\/Damken",
>             "storage": "ownCloud",
>             "authentication_type": "Username and password",
>             "configuration": {
>                 "host": "https:\/\/cloud.damken.com",
>                 "root": "\/",
>                 "secure": true,
>                 "user": "password",
>                 "password": "password"
>             },
>             "options": {
>                 "encrypt": true,
>                 "previews": true,
>                 "enable_sharing": false,
>                 "filesystem_check_changes": 1,
>                 "encoding_compatibility": false
>             },
>             "applicable_users": "All",
>             "applicable_groups": "",
>             "type": "Admin"
root@cloudine:/var/www/html/owncloud#
```

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -39,7 +39,8 @@ private function registerServices() {
 				\OC_Util::getEditionString(),
 				\OCP\User::getDisplayName(),
 				\OC::$server->getSystemConfig(),
-				\OC::$server->getAppConfig()
+				\OC::$server->getAppConfig(),
+                \OC::$server->getGlobalStoragesService()
```
Comment (PVince81) [inline lib/AppInfo/Application.php]: slipped

File: lib/ReportDataCollector.php [hunk 2]
```diff
@@ -143,10 +153,45 @@ public function getReport() {
...
 			'integritychecker' => $this->getIntegrityCheckerDetailArray(),
 			'apps' => $this->getAppsDetailArray(),
+            'mounts' => $this->getMountsArray(),
 			'phpinfo' => $this->getPhpInfoDetailArray()
 		];
 	}
 
+    /**
+     * @return array
+     */
```
Comment (PVince81) [inline lib/ReportDataCollector.php]: slipped?

File: lib/ReportDataCollector.php [hunk 3]
```diff
@@ -143,10 +153,45 @@ public function getReport() {
 			'config' => $this->getSystemConfigDetailArray(),
 			'integritychecker' => $this->getIntegrityCheckerDetailArray(),
 			'apps' => $this->getAppsDetailArray(),
+            'mounts' => $this->getMountsArray(),
```
Comment (PVince81) [inline lib/ReportDataCollector.php]: ohhhhh

**Comment by cdamken** [comment]
@PVince81 We need to have in the config-report all mounted external storages to see if their configuration is correct.

**Comment by sharidas** [comment]
Updated this PR. Verified the output by adding SFTP and owncloud external storages. Here is the output of the `mounts` from the globalstorages:
```
"mounts": [
        {
            "id": 1,
            "mount_point": "\/SFTP",
            "storage": "SFTP",
            "authentication_type": "Username and password",
            "configuration": {
                "host": "localhost",
                "root": "\/tmp\/foo",
                "user": "sujith",
                "password": "***REMOVED SENSITIVE VALUE***"
            },
            "options": {
                "encrypt": true,
                "previews": true,
                "enable_sharing": false,
                "filesystem_check_changes": 1,
                "encoding_compatibility": false
            },
            "applicable_users": "user1",
            "applicable_groups": "",
            "type": "Admin"
        },
        {
            "id": 2,
            "mount_point": "\/SFTP1",
            "storage": "SFTP",
            "authentication_type": "Username and password",
            "configuration": {
                "host": "localhost",
                "root": "\/tmp\/bar",
                "user": "sujith",
                "password": "***REMOVED SENSITIVE VALUE***"
            },
            "options": {
                "encrypt": true,
                "previews": true,
                "enable_sharing": false,
                "filesystem_check_changes": 1,
                "encoding_compatibility": false
            },
            "applicable_users": "user2",
            "applicable_groups": "",
            "type": "Admin"
        },
        {
            "id": 3,
            "mount_point": "\/ownCloud",
            "storage": "ownCloud",
            "authentication_type": "Log-in credentials, save in session",
            "configuration": {
                "host": "http:\/\/localhost\/testing2",
                "root": "",
                "secure": false,
                "user": "admin",
                "password": "***REMOVED SENSITIVE VALUE***"
            },
            "options": {
                "encrypt": true,
                "previews": true,
                "enable_sharing": false,
                "filesystem_check_changes": 1,
                "encoding_compatibility": false
            },
            "applicable_users": "admin, user3",
            "applicable_groups": "",
            "type": "Admin"
        }
    ],
```

Let me know if this looks ok.

**Comment by PVince81** [comment]
@sharidas please fix CI failures

**Comment by sharidas** [comment]
Merging it as the CI is green.

**Comment by sharidas** [comment]
Backport PR : https://github.com/owncloud/configreport/pull/94

**Pull request merged**